entitlement/src/main/java/org/killbill/billing/entitlement/api/DefaultEntitlementApi.java

```diff
@@ -156,6 +157,58 @@ public Entitlement doCall(final EntitlementApi entitlementApi, final Entitlement
     }
 
     @Override
+    public Entitlement createBaseEntitlementWithAddOns(final UUID accountId, final Iterable<EntitlementSpecifier> entitlementSpecifier, final LocalDate effectiveDate,
+                                                             final Iterable<PluginProperty> properties, final CallContext callContext) throws EntitlementApiException {
+
+        final EntitlementSpecifier baseSpecifier = Iterables.tryFind(entitlementSpecifier, new Predicate<EntitlementSpecifier>() {
+            @Override
+            public boolean apply(final EntitlementSpecifier specifier) {
+                return specifier.getPlanPhaseSpecifier() != null && ProductCategory.BASE.equals(specifier.getPlanPhaseSpecifier().getProductCategory());
+            }
+        }).orNull();
+
+        if (baseSpecifier == null) {
+            throw new EntitlementApiException(new IllegalArgumentException(), ErrorCode.SUB_CREATE_NO_BP.getCode(), "Missing Base Subscription.");
+        }
+
+        final EntitlementContext pluginContext = new DefaultEntitlementContext(OperationType.CREATE_SUBSCRIPTION,
+                                                                               accountId,
```

Comment:
@maguero Your concern is valid, the change in not backward compatible. This entitlement plugin api is not very stable (i think nobody uses it today but us internally). So i would proceed with the change:
1. Create new type `OperationType` `CREATE_SUBSCRIPTIONS_WITH_AO `
2. Modify context to pass a list of  `List<EntitlementSpecifier>` to reflect the multiple subscriptions (ideally BASE should be first entry).